src/apis/web3api.py

```diff
@@ -0,0 +1,215 @@
+"""
+Web3API for fetching relevant data using the web3 library.
+"""
+from os import getenv
+from typing import Dict, List, Tuple, Any, Optional
+from fractions import Fraction
+from dotenv import load_dotenv
+from web3 import Web3
+from web3.types import TxData, TxReceipt
+from eth_typing import Address, HexStr
+from hexbytes import HexBytes
+from contracts.gpv2_settlement import gpv2_settlement as gpv2Abi
+from src.trades import Trade, OrderData, OrderExecution
+from src.helper_functions import get_logger
+from src.constants import (
+    ADDRESS,
+)
```

Comment:
nit -- this does not appear to need multiline import (at least not for now).
```suggestion
from src.constants import ADDRESS
```

Reply:
Also, I think ADDRESS is not a sufficient enough name for me to know "what address is it?". Is it my home address? Based on how its used below, it appears to be the `SETTLEMENT_CONTRACT_ADDRESS`